backend/app/main/crud.py

```diff
@@ -147,6 +154,36 @@ def remove_user(session: Session, user_id: int):
     return user_id
 
 
+def verify_password(plain_password, hashed_password):
+    return pwd_context.verify(plain_password, hashed_password)
+
+
+def authenticate_user(session: Session, username: str, password: str):
+    user = get_user(session, username)
+    if not user:
+        return False
+    if not verify_password(password, user.password):
+        return False
+    return user
+
+
+# to get a string like this run:
+# openssl rand -hex 32
+SECRET_KEY = "0c4e63c7dc5c9aa52b419a1b1637af973744898204dd18ebcc0632a78ef051d0"
+ALGORITHM = "HS256"
```

Comment:
Hey alice great work, we shouldnt' include any secret keys in the PR. Please look into creating a .env file that is something hidden from git. Something similar is written in here: https://fastapi.tiangolo.com/advanced/settings/#read-env-vars-in-python except they just export the variable through the command line (You could do, but I suggest creating a file). To create an environment file, you would create a hidden .env file and put it in the gitignore. Once done, your code will have to target that file to extract variables in there. Check this out: https://fastapi.tiangolo.com/advanced/settings/#reading-a-env-file

Reply:
fixed with the .env file! pls check